lib/src/main/java/com/otaliastudios/transcoder/internal/thumbnails/DefaultThumbnailsEngine.kt

```diff
@@ -121,15 +123,18 @@ class DefaultThumbnailsEngine(
                 val nextKeyFrameUs = source.keyFrameAt(nextKeyFrameIndex) { Long.MAX_VALUE }
                 val previousKeyFrameUs = source.keyFrameAt(nextKeyFrameIndex - 1) { source.lastKeyFrame() }
 
-                log.i(
-                    "seek: current ${source.positionUs}," +
-                        " requested $requested, threshold $threshold, nextKeyFrameUs $nextKeyFrameUs"
-                )
 
                 val rightGap = nextKeyFrameUs - requested
                 val nextKeyFrameInThreshold = rightGap <= threshold
                 seek = nextKeyFrameInThreshold || previousKeyFrameUs > current || (current - requested > threshold)
-                seekUs = if (nextKeyFrameInThreshold) nextKeyFrameUs else previousKeyFrameUs
+                seekUs =
+                    (if (nextKeyFrameInThreshold) nextKeyFrameUs else previousKeyFrameUs) + source.seekThreshold
```

Comment:
why add seekThreshold here?

Reply:
same reason why we added it in data source.
i want to seek to 8 seconds which is my keyframe.
turns out, giving exact 8 seconds as value, it seeks to the previous keyframe, and not 8 itself. so this threshold is needed